libfreerdp/core/client.c

```diff
@@ -570,25 +572,27 @@ BOOL freerdp_channels_process_message_free(wMessage* message, DWORD type)
 
 static BOOL freerdp_channels_process_message(freerdp* instance, wMessage* message)
 {
-	if (message->id == WMQ_QUIT)
-	{
-		return FALSE;
-	}
+	BOOL rc = FALSE;
 
-	if (message->id == 0)
+	WINPR_ASSERT(instance);
+	WINPR_ASSERT(message);
+
+	if (message->id == WMQ_QUIT)
+		rc = FALSE;
```

Comment:
This is confusing - should this actually be `goto fail;` (and return `FALSE`) or really `rc = FALSE;` which gets overriden to `TRUE` later on (and finally return `TRUE`)?

Reply:
good catch ^^